src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -390,27 +460,34 @@ private void DisplayTemplateList(bool showAll = false)
 
             foreach (IGrouping<string, ITemplateInfo> grouping in grouped)
             {
-                List<string> languages = new List<string>();
+                List<string> languageForDisplay = new List<string>();
+                HashSet<string> uniqueLanguages = new HashSet<string>();
 
                 foreach (ITemplateInfo info in grouping)
                 {
                     if (info.Tags != null && info.Tags.TryGetValue("language", out ICacheTag languageTag))
                     {
                         foreach (string lang in languageTag.ChoicesAndDescriptions.Keys)
                         {
-                            if (string.IsNullOrEmpty(Language) && string.Equals(defaultLanguage, lang, StringComparison.OrdinalIgnoreCase))
+                            if (!uniqueLanguages.Contains(lang))
                             {
-                                languages.Add($"[{lang}]");
-                            }
-                            else
-                            {
-                                languages.Add(lang);
+                                if (string.IsNullOrEmpty(Language) && string.Equals(defaultLanguage, lang, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    languageForDisplay.Add($"[{lang}]");
+                                }
+                                else
+                                {
+                                    languageForDisplay.Add(lang);
+                                    uniqueLanguages.Add(lang);
+                                }
+
+                                uniqueLanguages.Add(lang);
```

Comment:
This add happens both in the else and outside of it - the one in the else can probably be removed, as could this one if the initial contains check is replaced with an Add